cnd/src/btsieve/mod.rs

```diff
@@ -37,6 +41,19 @@ pub trait BlockByHash: Send + Sync + 'static {
     ) -> Box<dyn Future<Item = Self::Block, Error = Self::Error> + Send + 'static>;
 }
 
+#[async_trait]
+pub trait BlockCache: Send + Sync + 'static {
```

Comment:
There is no point in requiring the caller to be sync if you have a method with `&mut`. How does this even compile? :D